src/ext/respToRedisLoader.c

```diff
@@ -323,16 +315,23 @@ _LIBRDB_API RdbxRespToRedisLoader *RDBX_createRespToRedisTcp(RdbParser *p,
     if (inet_pton(AF_INET, hostname, &(server_addr.sin_addr)) <= 0) {
         RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_INVALID_ADDRESS,
                         "Invalid tcp address (hostname=%s, port=%d)", hostname, port);
-        close(sockfd);
-        return NULL;
+        goto createErr;
     }
 
     if (connect(sockfd, (struct sockaddr *) &server_addr, sizeof(server_addr)) == -1) {
         RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_INVALID_ADDRESS,
                         "Invalid tcp address (hostname=%s, port=%d)", hostname, port);
-        close(sockfd);
-        return NULL;
+        goto createErr;
     }
 
-    return RDBX_createRespToRedisFd(p, rdbToResp, auth, sockfd);
+    RdbxRespToRedisLoader *res = RDBX_createRespToRedisFd(p, rdbToResp, auth, sockfd);
```

Comment:
I see inside RDBX_createRespToRedisFd(), it calls close() for fd in one error path. Probably, it is better to do it in a single place.

Reply:
right. thanks.